{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "machine_shape": "hm",
      "gpuType": "A100"
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "GPU"
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6giOFmSGaotX",
        "outputId": "a8bcaa60-3117-4809-a8dd-58a164f67ecd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'lag-llama'...\n",
            "remote: Enumerating objects: 486, done.\u001b[K\n",
            "remote: Counting objects: 100% (324/324), done.\u001b[K\n",
            "remote: Compressing objects: 100% (151/151), done.\u001b[K\n",
            "remote: Total 486 (delta 209), reused 260 (delta 169), pack-reused 162 (from 1)\u001b[K\n",
            "Receiving objects: 100% (486/486), 271.18 KiB | 15.07 MiB/s, done.\n",
            "Resolving deltas: 100% (250/250), done.\n"
          ]
        }
      ],
      "source": [
        "!git clone https://github.com/time-series-foundation-models/lag-llama.git\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%cd lag-llama\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "zciAcOQkastq",
        "outputId": "3f2195a5-3dec-4e3a-cbc2-ed4221b71b80"
      },
      "execution_count": 21,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content/lag-llama/lag-llama/lag-llama\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install -r requirements.txt\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "fIj0PzDuaxd6",
        "outputId": "e924b598-ed9a-428a-ee1e-6e2b4383db6c"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: gluonts<=0.14.4 in /usr/local/lib/python3.10/dist-packages (from gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (0.14.4)\n",
            "Requirement already satisfied: numpy>=1.23.5 in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 2)) (1.26.4)\n",
            "Requirement already satisfied: torch>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 3)) (2.5.1+cu121)\n",
            "Requirement already satisfied: wandb in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 4)) (0.18.7)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 5)) (1.13.1)\n",
            "Requirement already satisfied: pandas>=2.1.4 in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 6)) (2.1.4)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 8)) (3.8.0)\n",
            "Requirement already satisfied: huggingface_hub[cli] in /usr/local/lib/python3.10/dist-packages (from -r requirements.txt (line 7)) (0.26.2)\n",
            "Requirement already satisfied: pydantic<3,>=1.7 in /usr/local/lib/python3.10/dist-packages (from gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (2.9.2)\n",
            "Requirement already satisfied: tqdm~=4.23 in /usr/local/lib/python3.10/dist-packages (from gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (4.66.6)\n",
            "Requirement already satisfied: toolz~=0.10 in /usr/local/lib/python3.10/dist-packages (from gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (0.12.1)\n",
            "Requirement already satisfied: typing-extensions~=4.0 in /usr/local/lib/python3.10/dist-packages (from gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (4.12.2)\n",
            "Requirement already satisfied: filelock in /usr/local/lib/python3.10/dist-packages (from torch>=2.0.0->-r requirements.txt (line 3)) (3.16.1)\n",
            "Requirement already satisfied: networkx in /usr/local/lib/python3.10/dist-packages (from torch>=2.0.0->-r requirements.txt (line 3)) (3.4.2)\n",
            "Requirement already satisfied: jinja2 in /usr/local/lib/python3.10/dist-packages (from torch>=2.0.0->-r requirements.txt (line 3)) (3.1.4)\n",
            "Requirement already satisfied: fsspec in /usr/local/lib/python3.10/dist-packages (from torch>=2.0.0->-r requirements.txt (line 3)) (2024.10.0)\n",
            "Requirement already satisfied: sympy==1.13.1 in /usr/local/lib/python3.10/dist-packages (from torch>=2.0.0->-r requirements.txt (line 3)) (1.13.1)\n",
            "Requirement already satisfied: mpmath<1.4,>=1.1.0 in /usr/local/lib/python3.10/dist-packages (from sympy==1.13.1->torch>=2.0.0->-r requirements.txt (line 3)) (1.3.0)\n",
            "Requirement already satisfied: click!=8.0.0,>=7.1 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (8.1.7)\n",
            "Requirement already satisfied: docker-pycreds>=0.4.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (0.4.0)\n",
            "Requirement already satisfied: gitpython!=3.1.29,>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (3.1.43)\n",
            "Requirement already satisfied: platformdirs in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (4.3.6)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=5.28.0,<6,>=3.19.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (4.25.5)\n",
            "Requirement already satisfied: psutil>=5.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (5.9.5)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (6.0.2)\n",
            "Requirement already satisfied: requests<3,>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (2.32.3)\n",
            "Requirement already satisfied: sentry-sdk>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (2.18.0)\n",
            "Requirement already satisfied: setproctitle in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (1.3.4)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from wandb->-r requirements.txt (line 4)) (75.1.0)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.1.4->-r requirements.txt (line 6)) (2.8.2)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.1.4->-r requirements.txt (line 6)) (2024.2)\n",
            "Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.10/dist-packages (from pandas>=2.1.4->-r requirements.txt (line 6)) (2024.2)\n",
            "Requirement already satisfied: packaging>=20.9 in /usr/local/lib/python3.10/dist-packages (from huggingface_hub[cli]->-r requirements.txt (line 7)) (24.2)\n",
            "Requirement already satisfied: InquirerPy==0.3.4 in /usr/local/lib/python3.10/dist-packages (from huggingface_hub[cli]->-r requirements.txt (line 7)) (0.3.4)\n",
            "Requirement already satisfied: pfzy<0.4.0,>=0.3.1 in /usr/local/lib/python3.10/dist-packages (from InquirerPy==0.3.4->huggingface_hub[cli]->-r requirements.txt (line 7)) (0.3.4)\n",
            "Requirement already satisfied: prompt-toolkit<4.0.0,>=3.0.1 in /usr/local/lib/python3.10/dist-packages (from InquirerPy==0.3.4->huggingface_hub[cli]->-r requirements.txt (line 7)) (3.0.48)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (1.3.1)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (0.12.1)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (4.55.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (1.4.7)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (11.0.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.10/dist-packages (from matplotlib->-r requirements.txt (line 8)) (3.2.0)\n",
            "Requirement already satisfied: lightning<2.2,>=2.0 in /usr/local/lib/python3.10/dist-packages (from gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (2.1.4)\n",
            "Requirement already satisfied: pytorch-lightning<2.2,>=2.0 in /usr/local/lib/python3.10/dist-packages (from gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (2.1.4)\n",
            "Requirement already satisfied: six>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from docker-pycreds>=0.4.0->wandb->-r requirements.txt (line 4)) (1.16.0)\n",
            "Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.10/dist-packages (from gitpython!=3.1.29,>=1.0.0->wandb->-r requirements.txt (line 4)) (4.0.11)\n",
            "Requirement already satisfied: lightning-utilities<2.0,>=0.8.0 in /usr/local/lib/python3.10/dist-packages (from lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (0.11.9)\n",
            "Requirement already satisfied: torchmetrics<3.0,>=0.7.0 in /usr/local/lib/python3.10/dist-packages (from lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (1.6.0)\n",
            "Requirement already satisfied: annotated-types>=0.6.0 in /usr/local/lib/python3.10/dist-packages (from pydantic<3,>=1.7->gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (0.7.0)\n",
            "Requirement already satisfied: pydantic-core==2.23.4 in /usr/local/lib/python3.10/dist-packages (from pydantic<3,>=1.7->gluonts<=0.14.4->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (2.23.4)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 4)) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 4)) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 4)) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb->-r requirements.txt (line 4)) (2024.8.30)\n",
            "Requirement already satisfied: MarkupSafe>=2.0 in /usr/local/lib/python3.10/dist-packages (from jinja2->torch>=2.0.0->-r requirements.txt (line 3)) (3.0.2)\n",
            "Requirement already satisfied: aiohttp!=4.0.0a0,!=4.0.0a1 in /usr/local/lib/python3.10/dist-packages (from fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (3.11.2)\n",
            "Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.10/dist-packages (from gitdb<5,>=4.0.1->gitpython!=3.1.29,>=1.0.0->wandb->-r requirements.txt (line 4)) (5.0.1)\n",
            "Requirement already satisfied: wcwidth in /usr/local/lib/python3.10/dist-packages (from prompt-toolkit<4.0.0,>=3.0.1->InquirerPy==0.3.4->huggingface_hub[cli]->-r requirements.txt (line 7)) (0.2.13)\n",
            "Requirement already satisfied: aiohappyeyeballs>=2.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (2.4.3)\n",
            "Requirement already satisfied: aiosignal>=1.1.2 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (1.3.1)\n",
            "Requirement already satisfied: attrs>=17.3.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (24.2.0)\n",
            "Requirement already satisfied: frozenlist>=1.1.1 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (1.5.0)\n",
            "Requirement already satisfied: multidict<7.0,>=4.5 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (6.1.0)\n",
            "Requirement already satisfied: propcache>=0.2.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (0.2.0)\n",
            "Requirement already satisfied: yarl<2.0,>=1.17.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (1.17.2)\n",
            "Requirement already satisfied: async-timeout<6.0,>=4.0 in /usr/local/lib/python3.10/dist-packages (from aiohttp!=4.0.0a0,!=4.0.0a1->fsspec[http]<2025.0,>=2022.5.0->lightning<2.2,>=2.0->gluonts[torch]<=0.14.4->-r requirements.txt (line 1)) (4.0.3)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip list\n"
      ],
      "metadata": {
        "id": "x4KuZ6Ixay2Y"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!python --version\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bYiTTX-SbVz-",
        "outputId": "d05c6e62-c928-4e06-b31e-aa9140bbff0a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Python 3.10.12\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!gdown \"https://drive.google.com/uc?id=1JrDWMZyoPsc6d1wAAjgm3PosbGus-jCE\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Doodl5n4bi8p",
        "outputId": "342676d8-be71-433e-c6b8-8f30cb8c8011"
      },
      "execution_count": 23,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading...\n",
            "From (original): https://drive.google.com/uc?id=1JrDWMZyoPsc6d1wAAjgm3PosbGus-jCE\n",
            "From (redirected): https://drive.google.com/uc?id=1JrDWMZyoPsc6d1wAAjgm3PosbGus-jCE&confirm=t&uuid=080371d0-2e23-46cf-a44e-d3927e4cf24e\n",
            "To: /content/lag-llama/lag-llama/lag-llama/nonmonash_datasets.tar.gz\n",
            "100% 166M/166M [00:00<00:00, 167MB/s]\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir -p datasets\n",
        "!tar -xvzf nonmonash_datasets.tar.gz -C datasets"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "QIfDwrs_b2Gl",
        "outputId": "df954182-9bd7-498e-e2e2-9aa98b9da3aa"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "air_quality/\n",
            "air_quality/beijing_pm25.json\n",
            "air_quality/beijing_multisite.json\n",
            "air_quality/AirQualityUCI.json\n",
            "ett_datasets/\n",
            "ett_datasets/.DS_Store\n",
            "ett_datasets/ett_h1/\n",
            "ett_datasets/ett_h1/.DS_Store\n",
            "ett_datasets/ett_h1/metadata.json\n",
            "ett_datasets/ett_h1/test/\n",
            "ett_datasets/ett_h1/test/.DS_Store\n",
            "ett_datasets/ett_h1/test/data.json.gz\n",
            "ett_datasets/ett_h1/train/\n",
            "ett_datasets/ett_h1/train/data.json.gz\n",
            "ett_datasets/ett_h1/train/modified_data.json.gz\n",
            "ett_datasets/ett_m2/\n",
            "ett_datasets/ett_m2/metadata.json\n",
            "ett_datasets/ett_m2/test/\n",
            "ett_datasets/ett_m2/test/data.json.gz\n",
            "ett_datasets/ett_m2/train/\n",
            "ett_datasets/ett_m2/train/data.json.gz\n",
            "ett_datasets/ett_h2/\n",
            "ett_datasets/ett_h2/metadata.json\n",
            "ett_datasets/ett_h2/test/\n",
            "ett_datasets/ett_h2/test/data.json.gz\n",
            "ett_datasets/ett_h2/train/\n",
            "ett_datasets/ett_h2/train/data.json.gz\n",
            "ett_datasets/ett_m1/\n",
            "ett_datasets/ett_m1/metadata.json\n",
            "ett_datasets/ett_m1/test/\n",
            "ett_datasets/ett_m1/test/data.json.gz\n",
            "ett_datasets/ett_m1/train/\n",
            "ett_datasets/ett_m1/train/data.json.gz\n",
            "huawei/\n",
            "huawei/.DS_Store\n",
            "huawei/cpu_limit_minute.json\n",
            "huawei/function_delay_minute.json\n",
            "huawei/instances_minute.json\n",
            "huawei/memory_limit_minute.json\n",
            "huawei/cpu_usage_minute.json\n",
            "huawei/memory_usage_minute.json\n",
            "huawei/platform_delay_minute.json\n",
            "huawei/requests_minute.json\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls datasets"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "2nIT_aqlb-56",
        "outputId": "6a83c998-dafb-4c0f-ddb7-b98da21a4d4f"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "air_quality  ett_datasets  huawei\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install wandb\n",
        "import wandb\n",
        "wandb.login()\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YGIfkCkJcF8e",
        "outputId": "bf78333d-5266-4803-d8ce-db0fef462860"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: wandb in /usr/local/lib/python3.10/dist-packages (0.18.7)\n",
            "Requirement already satisfied: click!=8.0.0,>=7.1 in /usr/local/lib/python3.10/dist-packages (from wandb) (8.1.7)\n",
            "Requirement already satisfied: docker-pycreds>=0.4.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (0.4.0)\n",
            "Requirement already satisfied: gitpython!=3.1.29,>=1.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (3.1.43)\n",
            "Requirement already satisfied: platformdirs in /usr/local/lib/python3.10/dist-packages (from wandb) (4.3.6)\n",
            "Requirement already satisfied: protobuf!=4.21.0,!=5.28.0,<6,>=3.19.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (4.25.5)\n",
            "Requirement already satisfied: psutil>=5.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (5.9.5)\n",
            "Requirement already satisfied: pyyaml in /usr/local/lib/python3.10/dist-packages (from wandb) (6.0.2)\n",
            "Requirement already satisfied: requests<3,>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (2.32.3)\n",
            "Requirement already satisfied: sentry-sdk>=2.0.0 in /usr/local/lib/python3.10/dist-packages (from wandb) (2.18.0)\n",
            "Requirement already satisfied: setproctitle in /usr/local/lib/python3.10/dist-packages (from wandb) (1.3.4)\n",
            "Requirement already satisfied: setuptools in /usr/local/lib/python3.10/dist-packages (from wandb) (75.1.0)\n",
            "Requirement already satisfied: typing-extensions<5,>=4.4 in /usr/local/lib/python3.10/dist-packages (from wandb) (4.12.2)\n",
            "Requirement already satisfied: six>=1.4.0 in /usr/local/lib/python3.10/dist-packages (from docker-pycreds>=0.4.0->wandb) (1.16.0)\n",
            "Requirement already satisfied: gitdb<5,>=4.0.1 in /usr/local/lib/python3.10/dist-packages (from gitpython!=3.1.29,>=1.0.0->wandb) (4.0.11)\n",
            "Requirement already satisfied: charset-normalizer<4,>=2 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb) (3.4.0)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb) (3.10)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb) (2.2.3)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.10/dist-packages (from requests<3,>=2.0.0->wandb) (2024.8.30)\n",
            "Requirement already satisfied: smmap<6,>=3.0.1 in /usr/local/lib/python3.10/dist-packages (from gitdb<5,>=4.0.1->gitpython!=3.1.29,>=1.0.0->wandb) (5.0.1)\n"
          ]
        },
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "True"
            ]
          },
          "metadata": {},
          "execution_count": 26
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import wandb\n",
        "wandb.login()"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BVffMVPHcyU5",
        "outputId": "57798cef-751e-4cb4-94ca-0df54cbea4b6"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "True"
            ]
          },
          "metadata": {},
          "execution_count": 27
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open(\"scripts/pretrain.sh\", \"w\") as f:\n",
        "    f.write(\"\"\"\n",
        "mkdir -p experiments\n",
        "mkdir -p experiments/seeds\n",
        "mkdir -p experiments/results\n",
        "\n",
        "EXP_NAME=\"pretraining_lag_llama\"\n",
        "FILENAME=\"experiments/seeds/${EXP_NAME}\"\n",
        "CONFIGPATH=\"configs/lag_llama.json\"\n",
        "\n",
        "echo $EXP_NAME\n",
        "\n",
        "# NUM_SEEDS used only if it is a new experiment\n",
        "NUM_SEEDS=1\n",
        "\n",
        "# Create seeds\n",
        "if [ -f $FILENAME ]; then\n",
        "    echo \"${FILENAME} already exists.\"\n",
        "\n",
        "    SEEDS=()\n",
        "    while read -r LINE; do\n",
        "        SEEDS+=(\"$LINE\")\n",
        "    done < $FILENAME\n",
        "    echo \"Found ${#SEEDS[@]} seeds for training.\"\n",
        "else\n",
        "    # Write seeds\n",
        "    echo \"${FILENAME} created. Writing seeds.\"\n",
        "    touch $FILENAME\n",
        "    for (( i = 0; i < $NUM_SEEDS; i++ ))\n",
        "    do\n",
        "        SEED=$((RANDOM + 1))\n",
        "        echo $SEED >> $FILENAME\n",
        "    done\n",
        "\n",
        "    # Read them\n",
        "    SEEDS=()\n",
        "    while read -r LINE; do\n",
        "        SEEDS+=(\"$LINE\")\n",
        "    done < $FILENAME\n",
        "fi\n",
        "\n",
        "# Train\n",
        "for SEED in \"${SEEDS[@]}\"\n",
        "do\n",
        "    EXPERIMENT_NAME=\"${EXP_NAME}_seed_${SEED}\"\n",
        "\n",
        "    python run.py \\\n",
        "    -e $EXP_NAME -d \"datasets\" --seed $SEED \\\n",
        "    -r \"experiments/results\" \\\n",
        "    --batch_size 512 -m 4 -n 128 \\\n",
        "    --wandb_entity \"j1zha-uc-san-diego\" --wandb_project \"lag_llama_experiment\" \\\n",
        "    --all_datasets \"weather\" \"platform_delay_minute\" \"requests_minute\"\\\n",
        "    --test_datasets \"weather\"\\\n",
        "    --num_workers 2 --args_from_dict_path $CONFIGPATH --search_batch_size \\\n",
        "    --lr 0.0001\n",
        "done\n",
        "    \"\"\")\n"
      ],
      "metadata": {
        "id": "tDfVeNjTgern"
      },
      "execution_count": 39,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!cat scripts/pretrain.sh\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "07RxUx5kIDDk",
        "outputId": "b305bbbb-55b8-453f-ba30-3b48d2867506"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "mkdir -p experiments\n",
            "mkdir -p experiments/seeds\n",
            "mkdir -p experiments/results\n",
            "\n",
            "EXP_NAME=\"pretraining_lag_llama\"\n",
            "FILENAME=\"experiments/seeds/${EXP_NAME}\"\n",
            "CONFIGPATH=\"configs/lag_llama.json\"\n",
            "\n",
            "echo $EXP_NAME\n",
            "\n",
            "# NUM_SEEDS used only if it is a new experiment\n",
            "NUM_SEEDS=1\n",
            "\n",
            "# Create seeds\n",
            "if [ -f $FILENAME ]; then\n",
            "    echo \"${FILENAME} already exists.\"\n",
            "\n",
            "    SEEDS=()\n",
            "    while read -r LINE; do\n",
            "        SEEDS+=(\"$LINE\")\n",
            "    done < $FILENAME\n",
            "    echo \"Found ${#SEEDS[@]} seeds for training.\"\n",
            "else\n",
            "    # Write seeds\n",
            "    echo \"${FILENAME} created. Writing seeds.\"\n",
            "    touch $FILENAME\n",
            "    for (( i = 0; i < $NUM_SEEDS; i++ )) \n",
            "    do \n",
            "        SEED=$((RANDOM + 1))\n",
            "        echo $SEED >> $FILENAME\n",
            "    done\n",
            "\n",
            "    # Read them\n",
            "    SEEDS=()\n",
            "    while read -r LINE; do\n",
            "        SEEDS+=(\"$LINE\")\n",
            "    done < $FILENAME\n",
            "fi\n",
            "\n",
            "# Train\n",
            "for SEED in \"${SEEDS[@]}\"\n",
            "do\n",
            "    EXPERIMENT_NAME=\"${EXP_NAME}_seed_${SEED}\"\n",
            "\n",
            "    python run.py     -e $EXP_NAME -d \"datasets\" --seed $SEED     -r \"experiments/results\"     --batch_size 512 -m 4 -n 128     --wandb_entity \"j1zha-uc-san-diego\" --wandb_project \"lag_llama_experiment\"     --all_datasets \"weather\" \"platform_delay_minute\" \"requests_minute\"    --test_datasets \"weather\"    --num_workers 2 --args_from_dict_path $CONFIGPATH --search_batch_size     --lr 0.0001\n",
            "done\n",
            "    "
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!bash scripts/pretrain.sh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "NbQHT2FXEnOx",
        "outputId": "7f1fbeec-5505-4bcc-bdea-d4b834ef705f"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "pretraining_lag_llama\n",
            "experiments/seeds/pretraining_lag_llama already exists.\n",
            "Found 1 seeds for training.\n",
            "experiment_name : pretraining_lag_llama\n",
            "dataset_path : datasets\n",
            "all_datasets : ['weather', 'platform_delay_minute', 'requests_minute']\n",
            "test_datasets : ['weather']\n",
            "stratified_sampling : series\n",
            "seed : 27049\n",
            "context_length : 32\n",
            "prediction_length : 1\n",
            "max_prediction_length : 1024\n",
            "n_layer : 8\n",
            "num_encoder_layer : 4\n",
            "n_embd_per_head : 16\n",
            "n_head : 9\n",
            "dim_feedforward : 256\n",
            "lags_seq : ['Q', 'M', 'W', 'D', 'H', 'T', 'S']\n",
            "data_normalization : robust\n",
            "aug_prob : 0.5\n",
            "freq_mask_rate : 0.5\n",
            "freq_mixing_rate : 0.25\n",
            "jitter_prob : 0\n",
            "jitter_sigma : 0.03\n",
            "scaling_prob : 0\n",
            "scaling_sigma : 0.1\n",
            "rotation_prob : 0\n",
            "permutation_prob : 0\n",
            "permutation_max_segments : 5\n",
            "permutation_seg_mode : equal\n",
            "magnitude_warp_prob : 0\n",
            "magnitude_warp_sigma : 0.2\n",
            "magnitude_warp_knot : 4\n",
            "time_warp_prob : 0\n",
            "time_warp_sigma : 0.2\n",
            "time_warp_knot : 4\n",
            "window_slice_prob : 0\n",
            "window_slice_reduce_ratio : 0.9\n",
            "window_warp_prob : 0\n",
            "window_warp_window_ratio : 0.1\n",
            "window_warp_scales : [0.5, 2.0]\n",
            "time_feat : True\n",
            "batch_size : 512\n",
            "max_epochs : 4\n",
            "num_batches_per_epoch : 128\n",
            "limit_val_batches : None\n",
            "early_stopping_patience : 50\n",
            "dropout : 0.0\n",
            "num_parallel_samples : 100\n",
            "num_samples : 100\n",
            "num_workers : 2\n",
            "gpu : 0\n",
            "results_dir : experiments/results\n",
            "wandb_entity : j1zha-uc-san-diego\n",
            "wandb_project : lag_llama_experiment\n",
            "wandb_tags : None\n",
            "wandb_mode : online\n",
            "evaluate_only : False\n",
            "use_kv_cache : True\n",
            "swa : False\n",
            "swa_lrs : 0.01\n",
            "swa_epoch_start : 0.8\n",
            "annealing_epochs : 10\n",
            "annealing_strategy : cos\n",
            "use_single_instance_sampler : True\n",
            "plot_test_forecasts : True\n",
            "search_batch_size : True\n",
            "num_validation_windows : 14\n",
            "lr : 0.0001\n",
            "weight_decay : 0.0\n",
            "args_from_dict_path : configs/lag_llama.json\n",
            "eval_prefix : None\n",
            "ckpt_path : None\n",
            "get_ckpt_path_from_experiment_name : None\n",
            "single_dataset : None\n",
            "use_dataset_prediction_length : False\n",
            "single_dataset_last_k_percentage : None\n",
            "use_cosine_annealing_lr : False\n",
            "cosine_annealing_lr_t_max : 10000\n",
            "cosine_annealing_lr_eta_min : 0.01\n",
            "distr_output : studentT\n",
            "Seed set to 27049\n",
            "experiments/results/pretraining_lag_llama/27049 already exists.\n",
            "Checkpoint experiments/results/pretraining_lag_llama/27049/checkpoints/last.ckpt retrieved from experiment directory\n",
            "Training datasets: ['platform_delay_minute', 'requests_minute']\n",
            "Test datasets: ['weather']\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Currently logged in as: \u001b[33mj1zha\u001b[0m (\u001b[33mj1zha-uc-san-diego\u001b[0m). Use \u001b[1m`wandb login --relogin`\u001b[0m to force relogin\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Tracking run with wandb version 0.18.7\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run data is saved locally in \u001b[35m\u001b[1mexperiments/results/pretraining_lag_llama/27049/wandb/run-20241202_041059-3056ea2a\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run \u001b[1m`wandb offline`\u001b[0m to turn off syncing.\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Resuming run \u001b[33mpretraining_lag_llama-seed-27049\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: ⭐️ View project at \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 🚀 View run at \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment/runs/3056ea2a\u001b[0m\n",
            "Context length: 32 Prediction Length: 1 max(lags_seq): 1092 Therefore, window size: 1125\n",
            "Dataset: platform_delay_minute Total train points: 647860 Total val points: 11390\n",
            "Dataset: requests_minute Total train points: 647860 Total val points: 11390\n",
            "Trying batch size: 512\n",
            "GPU available: True (cuda), used: True\n",
            "TPU available: False, using: 0 TPU cores\n",
            "IPU available: False, using: 0 IPUs\n",
            "HPU available: False, using: 0 HPUs\n",
            "You are using a CUDA device ('NVIDIA A100-SXM4-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision\n",
            "2024-12-02 04:11:01.893624: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.\n",
            "2024-12-02 04:11:01.909082: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "2024-12-02 04:11:01.926754: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "2024-12-02 04:11:01.931930: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "2024-12-02 04:11:01.946493: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "2024-12-02 04:11:03.224148: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
            "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
            "\n",
            "  | Name          | Type               | Params\n",
            "-----------------------------------------------------\n",
            "0 | model         | LagLlamaModel      | 2.4 M \n",
            "1 | augmentations | ApplyAugmentations | 0     \n",
            "-----------------------------------------------------\n",
            "2.4 M     Trainable params\n",
            "0         Non-trainable params\n",
            "2.4 M     Total params\n",
            "9.797     Total estimated model params size (MB)\n",
            "Epoch 0: |          | 0/? [00:00<?, ?it/s] \n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation DataLoader 0: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Epoch 0: |          | 0/? [02:19<?, ?it/s, v_num=2]Epoch 0, global step 10: 'val_loss' reached 0.38012 (best 0.38012), saving model to 'experiments/results/pretraining_lag_llama/27049/batch-size-search/lightning_logs/version_2/checkpoints/epoch=0-step=10.ckpt' as top 1\n",
            "`Trainer.fit` stopped: `max_epochs=1` reached.\n",
            "Epoch 0: |          | 0/? [02:19<?, ?it/s, v_num=2]\n",
            "\n",
            "Using a batch size of 256 \n",
            "\n",
            "GPU available: True (cuda), used: True\n",
            "TPU available: False, using: 0 TPU cores\n",
            "IPU available: False, using: 0 IPUs\n",
            "HPU available: False, using: 0 HPUs\n",
            "Restoring states from the checkpoint path at experiments/results/pretraining_lag_llama/27049/checkpoints/last.ckpt\n",
            "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
            "\n",
            "  | Name          | Type               | Params\n",
            "-----------------------------------------------------\n",
            "0 | model         | LagLlamaModel      | 2.4 M \n",
            "1 | augmentations | ApplyAugmentations | 0     \n",
            "-----------------------------------------------------\n",
            "2.4 M     Trainable params\n",
            "0         Non-trainable params\n",
            "2.4 M     Total params\n",
            "9.797     Total estimated model params size (MB)\n",
            "Restored all states from the checkpoint at experiments/results/pretraining_lag_llama/27049/checkpoints/last.ckpt\n",
            "`Trainer.fit` stopped: `max_epochs=4` reached.\n",
            "Using checkpoint: experiments/results/pretraining_lag_llama/27049/lag_llama_experiment/3056ea2a/checkpoints/epoch=3-step=512.ckpt for evaluation\n",
            "Evaluating on weather\n",
            "# of Series in the test data: 3010\n",
            "Trying batch size: 256\n",
            "Plotting forecasts\n",
            "Running evaluation: 100% 3010/3010 [00:00<00:00, 643862.46it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:105: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return np.mean(np.abs(target - forecast)) / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:154: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return numerator / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:105: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return np.mean(np.abs(target - forecast)) / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:154: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return numerator / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "Evaluating on platform_delay_minute\n",
            "# of Series in the test data: 10\n",
            "Trying batch size: 256\n",
            "Plotting forecasts\n",
            "Running evaluation: 100% 10/10 [00:00<00:00, 3980.93it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "Evaluating on requests_minute\n",
            "# of Series in the test data: 10\n",
            "Trying batch size: 256\n",
            "Plotting forecasts\n",
            "Running evaluation: 100% 10/10 [00:00<00:00, 3775.25it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.028 MB of 0.028 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.036 MB of 0.036 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.037 MB of 0.037 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.037 MB of 0.037 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                                                                                \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.037 MB of 0.037 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run history:\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                  num_parameters ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: test/platform_delay_minute/CRPS ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:       test/requests_minute/CRPS ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:               test/weather/CRPS ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:             trainer/global_step ▁▁▁▁▁▁▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run summary:\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                                   epoch 4\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                                 lr-Adam 0.0001\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                          num_parameters 2449299\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: test/australian_electricity_demand/CRPS 0.21998\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                  test/beijing_pm25/CRPS 0.16621\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:            test/electricity_hourly/CRPS 0.22028\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                        test/ett_m2/CRPS 0.07745\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                 test/exchange_rate/CRPS 0.02177\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:             test/pedestrian_counts/CRPS 0.76247\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:         test/platform_delay_minute/CRPS 0.12017\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:               test/requests_minute/CRPS 0.1014\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                       test/weather/CRPS 0.20376\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                              train_loss -0.36471\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                     trainer/global_step 0\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                                val_loss -1.17549\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 🚀 View run \u001b[33mpretraining_lag_llama-seed-27049\u001b[0m at: \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment/runs/3056ea2a\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: ⭐️ View project at: \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Synced 4 W&B file(s), 0 media file(s), 3 artifact file(s) and 0 other file(s)\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Find logs at: \u001b[35m\u001b[1mexperiments/results/pretraining_lag_llama/27049/wandb/run-20241202_041059-3056ea2a/logs\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "with open(\"scripts/finetune.sh\", \"w\") as f:\n",
        "    f.write(\"\"\"\n",
        "    CONFIGPATH=\"configs/lag_llama.json\"\n",
        "PRETRAINING_EXP_NAME=\"pretraining_lag_llama\"\n",
        "PERCENTAGE=80\n",
        "\n",
        "for FINETUNE_DATASET in \"weather\"\n",
        "do\n",
        "    EXP_NAME=\"${PRETRAINING_EXP_NAME}_finetune_on_${FINETUNE_DATASET}\"\n",
        "\n",
        "    # We reuse the same seeds as used for pretraining\n",
        "    FILENAME=\"experiments/seeds/${PRETRAINING_EXP_NAME}\"\n",
        "    echo $PRETRAINING_EXP_NAME\n",
        "\n",
        "    # Get the seeds\n",
        "    if [ -f $FILENAME ]; then\n",
        "        echo \"${FILENAME} found. Reading seeds.\"\n",
        "        SEEDS=()\n",
        "        while read -r LINE; do\n",
        "            SEEDS+=(\"$LINE\")\n",
        "        done < $FILENAME\n",
        "        echo \"Found ${#SEEDS[@]} seeds for finetuning.\"\n",
        "    else\n",
        "        echo \"${FILENAME} does not exist. Cannot perform finetuning.\"\n",
        "        exit 0\n",
        "    fi\n",
        "\n",
        "    # Iterate through all training dataset\n",
        "    for SEED in \"${SEEDS[@]}\"\n",
        "    do\n",
        "        EXPERIMENT_NAME=\"${EXP_NAME}_seed_${SEED}\"\n",
        "\n",
        "        python run.py \\\n",
        "        -e $EXPERIMENT_NAME -d \"datasets\" --seed $SEED \\\n",
        "        -r \"experiments/results\" \\\n",
        "        --batch_size 512 -m 6 -n 128 \\\n",
        "        --wandb_entity \"j1zha-uc-san-diego\" --wandb_project \"lag_llama_experiment\" \\\n",
        "        --num_workers 2 --args_from_dict_path $CONFIGPATH --search_batch_size \\\n",
        "        --single_dataset $FINETUNE_DATASET \\\n",
        "        --get_ckpt_path_from_experiment_name $PRETRAINING_EXP_NAME --lr 0.00001 --use_dataset_prediction_length --num_validation_windows 1 \\\n",
        "        --single_dataset_last_k_percentage $PERCENTAGE\n",
        "    done\n",
        "done\n",
        "     \"\"\")"
      ],
      "metadata": {
        "id": "6doa0NnNHyqz"
      },
      "execution_count": 62,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "!bash scripts/finetune.sh"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ss5Z2oJnnbrN",
        "outputId": "a8c36897-255c-40fe-e295-54b929a2d131"
      },
      "execution_count": 63,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "pretraining_lag_llama\n",
            "experiments/seeds/pretraining_lag_llama found. Reading seeds.\n",
            "Found 1 seeds for finetuning.\n",
            "experiment_name : pretraining_lag_llama_finetune_on_weather_seed_27049\n",
            "dataset_path : datasets\n",
            "all_datasets : ['australian_electricity_demand', 'electricity_hourly', 'london_smart_meters_without_missing', 'solar_10_minutes', 'wind_farms_without_missing', 'pedestrian_counts', 'uber_tlc_hourly', 'traffic', 'kdd_cup_2018_without_missing', 'saugeenday', 'sunspot_without_missing', 'exchange_rate', 'cpu_limit_minute', 'cpu_usage_minute', 'function_delay_minute', 'instances_minute', 'memory_limit_minute', 'memory_usage_minute', 'platform_delay_minute', 'requests_minute', 'ett_h1', 'ett_h2', 'ett_m1', 'ett_m2', 'beijing_pm25', 'AirQualityUCI', 'beijing_multisite']\n",
            "test_datasets : []\n",
            "stratified_sampling : series\n",
            "seed : 27049\n",
            "context_length : 32\n",
            "prediction_length : 1\n",
            "max_prediction_length : 1024\n",
            "n_layer : 8\n",
            "num_encoder_layer : 4\n",
            "n_embd_per_head : 16\n",
            "n_head : 9\n",
            "dim_feedforward : 256\n",
            "lags_seq : ['Q', 'M', 'W', 'D', 'H', 'T', 'S']\n",
            "data_normalization : robust\n",
            "aug_prob : 0.5\n",
            "freq_mask_rate : 0.5\n",
            "freq_mixing_rate : 0.25\n",
            "jitter_prob : 0\n",
            "jitter_sigma : 0.03\n",
            "scaling_prob : 0\n",
            "scaling_sigma : 0.1\n",
            "rotation_prob : 0\n",
            "permutation_prob : 0\n",
            "permutation_max_segments : 5\n",
            "permutation_seg_mode : equal\n",
            "magnitude_warp_prob : 0\n",
            "magnitude_warp_sigma : 0.2\n",
            "magnitude_warp_knot : 4\n",
            "time_warp_prob : 0\n",
            "time_warp_sigma : 0.2\n",
            "time_warp_knot : 4\n",
            "window_slice_prob : 0\n",
            "window_slice_reduce_ratio : 0.9\n",
            "window_warp_prob : 0\n",
            "window_warp_window_ratio : 0.1\n",
            "window_warp_scales : [0.5, 2.0]\n",
            "time_feat : True\n",
            "batch_size : 512\n",
            "max_epochs : 6\n",
            "num_batches_per_epoch : 128\n",
            "limit_val_batches : None\n",
            "early_stopping_patience : 50\n",
            "dropout : 0.0\n",
            "num_parallel_samples : 100\n",
            "num_samples : 100\n",
            "num_workers : 2\n",
            "gpu : 0\n",
            "results_dir : experiments/results\n",
            "wandb_entity : j1zha-uc-san-diego\n",
            "wandb_project : lag_llama_experiment\n",
            "wandb_tags : None\n",
            "wandb_mode : online\n",
            "evaluate_only : False\n",
            "use_kv_cache : True\n",
            "swa : False\n",
            "swa_lrs : 0.01\n",
            "swa_epoch_start : 0.8\n",
            "annealing_epochs : 10\n",
            "annealing_strategy : cos\n",
            "use_single_instance_sampler : True\n",
            "plot_test_forecasts : True\n",
            "search_batch_size : True\n",
            "num_validation_windows : 1\n",
            "lr : 1e-05\n",
            "weight_decay : 0.0\n",
            "args_from_dict_path : configs/lag_llama.json\n",
            "eval_prefix : None\n",
            "ckpt_path : None\n",
            "get_ckpt_path_from_experiment_name : pretraining_lag_llama\n",
            "single_dataset : weather\n",
            "use_dataset_prediction_length : True\n",
            "single_dataset_last_k_percentage : 80.0\n",
            "use_cosine_annealing_lr : False\n",
            "cosine_annealing_lr_t_max : 10000\n",
            "cosine_annealing_lr_eta_min : 0.01\n",
            "distr_output : studentT\n",
            "Seed set to 27049\n",
            "experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049 already exists.\n",
            "Moving experiments/results/pretraining_lag_llama/27049/lag-llama/3056ea2a/checkpoints/last.ckpt to experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/checkpoints/pretrained_ckpt.ckpt\n",
            "Checkpoint experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/checkpoints/pretrained_ckpt.ckpt retrieved from experiment directory\n",
            "Training and test on weather\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Currently logged in as: \u001b[33mj1zha\u001b[0m (\u001b[33mj1zha-uc-san-diego\u001b[0m). Use \u001b[1m`wandb login --relogin`\u001b[0m to force relogin\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Tracking run with wandb version 0.18.7\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run data is saved locally in \u001b[35m\u001b[1mexperiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/wandb/run-20241202_045841-f32e2e0a\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run \u001b[1m`wandb offline`\u001b[0m to turn off syncing.\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Resuming run \u001b[33mpretraining_lag_llama_finetune_on_weather_seed_27049-seed-27049\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: ⭐️ View project at \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 🚀 View run at \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment/runs/f32e2e0a\u001b[0m\n",
            "Context length: 32 Prediction Length: 30 max(lags_seq): 1092 Therefore, window size: 1154\n",
            "Dataset: weather Total train points: 34352118 Total val points: 3476550\n",
            "Trying batch size: 512\n",
            "GPU available: True (cuda), used: True\n",
            "TPU available: False, using: 0 TPU cores\n",
            "IPU available: False, using: 0 IPUs\n",
            "HPU available: False, using: 0 HPUs\n",
            "You are using a CUDA device ('NVIDIA A100-SXM4-40GB') that has Tensor Cores. To properly utilize them, you should set `torch.set_float32_matmul_precision('medium' | 'high')` which will trade-off precision for performance. For more details, read https://pytorch.org/docs/stable/generated/torch.set_float32_matmul_precision.html#torch.set_float32_matmul_precision\n",
            "2024-12-02 04:58:55.127595: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.\n",
            "2024-12-02 04:58:55.142876: E external/local_xla/xla/stream_executor/cuda/cuda_fft.cc:485] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered\n",
            "2024-12-02 04:58:55.160370: E external/local_xla/xla/stream_executor/cuda/cuda_dnn.cc:8454] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered\n",
            "2024-12-02 04:58:55.165846: E external/local_xla/xla/stream_executor/cuda/cuda_blas.cc:1452] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered\n",
            "2024-12-02 04:58:55.181084: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.\n",
            "To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.\n",
            "2024-12-02 04:58:56.452925: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT\n",
            "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
            "\n",
            "  | Name          | Type               | Params\n",
            "-----------------------------------------------------\n",
            "0 | model         | LagLlamaModel      | 2.4 M \n",
            "1 | augmentations | ApplyAugmentations | 0     \n",
            "-----------------------------------------------------\n",
            "2.4 M     Trainable params\n",
            "0         Non-trainable params\n",
            "2.4 M     Total params\n",
            "9.797     Total estimated model params size (MB)\n",
            "Epoch 0: |          | 0/? [00:00<?, ?it/s] \n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation DataLoader 0: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Epoch 0: |          | 0/? [00:21<?, ?it/s, v_num=2]Epoch 0, global step 10: 'val_loss' reached 1.62632 (best 1.62632), saving model to 'experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/batch-size-search/lightning_logs/version_2/checkpoints/epoch=0-step=10.ckpt' as top 1\n",
            "`Trainer.fit` stopped: `max_epochs=1` reached.\n",
            "Epoch 0: |          | 0/? [00:21<?, ?it/s, v_num=2]\n",
            "\n",
            "Using a batch size of 256 \n",
            "\n",
            "GPU available: True (cuda), used: True\n",
            "TPU available: False, using: 0 TPU cores\n",
            "IPU available: False, using: 0 IPUs\n",
            "HPU available: False, using: 0 HPUs\n",
            "Restoring states from the checkpoint path at experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/checkpoints/pretrained_ckpt.ckpt\n",
            "LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [0]\n",
            "\n",
            "  | Name          | Type               | Params\n",
            "-----------------------------------------------------\n",
            "0 | model         | LagLlamaModel      | 2.4 M \n",
            "1 | augmentations | ApplyAugmentations | 0     \n",
            "-----------------------------------------------------\n",
            "2.4 M     Trainable params\n",
            "0         Non-trainable params\n",
            "2.4 M     Total params\n",
            "9.797     Total estimated model params size (MB)\n",
            "Restored all states from the checkpoint at experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/checkpoints/pretrained_ckpt.ckpt\n",
            "Epoch 5: |          | 120/? [02:06<00:00,  0.95it/s, v_num=2e0a]\n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Validation DataLoader 0: |          | 0/? [00:00<?, ?it/s]\u001b[A\n",
            "Epoch 5: |          | 120/? [02:15<00:00,  0.88it/s, v_num=2e0a]Metric val_loss improved. New best score: 1.532\n",
            "Epoch 5, global step 768: 'val_loss' reached 1.53219 (best 1.53219), saving model to 'experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/lag_llama_experiment/f32e2e0a/checkpoints/epoch=5-step=768-v1.ckpt' as top 1\n",
            "`Trainer.fit` stopped: `max_epochs=6` reached.\n",
            "Epoch 5: |          | 120/? [02:16<00:00,  0.88it/s, v_num=2e0a]\n",
            "Using checkpoint: experiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/lag_llama_experiment/f32e2e0a/checkpoints/epoch=5-step=768-v1.ckpt for evaluation\n",
            "Evaluating on weather\n",
            "# of Series in the test data: 3010\n",
            "Trying batch size: 256\n",
            "Plotting forecasts\n",
            "Running evaluation: 100% 3010/3010 [00:00<00:00, 627821.13it/s]\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:105: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return np.mean(np.abs(target - forecast)) / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/metrics.py:154: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  return numerator / seasonal_error\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "/usr/local/lib/python3.10/dist-packages/gluonts/evaluation/_base.py:425: RuntimeWarning: divide by zero encountered in scalar divide\n",
            "  metrics[\"ND\"] = cast(float, metrics[\"abs_error\"]) / cast(\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.024 MB of 0.024 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.029 MB of 0.029 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.029 MB of 0.029 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.029 MB of 0.029 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:                                                                                \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 0.029 MB of 0.029 MB uploaded\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run history:\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:               epoch ▁▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:             lr-Adam ▁▁▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:      num_parameters ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:   test/weather/CRPS ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:          train_loss ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: trainer/global_step ▁▇▇█████\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:            val_loss ▁\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Run summary:\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:               epoch 5\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:             lr-Adam 0.0001\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:      num_parameters 2449299\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:   test/weather/CRPS 0.1428\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:          train_loss 1.7346\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: trainer/global_step 767\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m:            val_loss 1.53219\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: \n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: 🚀 View run \u001b[33mpretraining_lag_llama_finetune_on_weather_seed_27049-seed-27049\u001b[0m at: \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment/runs/f32e2e0a\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: ⭐️ View project at: \u001b[34m\u001b[4mhttps://wandb.ai/j1zha-uc-san-diego/lag_llama_experiment\u001b[0m\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Synced 5 W&B file(s), 0 media file(s), 2 artifact file(s) and 0 other file(s)\n",
            "\u001b[34m\u001b[1mwandb\u001b[0m: Find logs at: \u001b[35m\u001b[1mexperiments/results/pretraining_lag_llama_finetune_on_weather_seed_27049/27049/wandb/run-20241202_045841-f32e2e0a/logs\u001b[0m\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "\n",
        "path = \"experiments/results/pretraining_lag_llama/27049/checkpoints/\"\n",
        "if os.path.exists(path):\n",
        "    print(\"Directory exists. Contents:\")\n",
        "    print(os.listdir(path))\n",
        "else:\n",
        "    print(\"Directory does not exist.\")\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Kb_zSw4wnqpe",
        "outputId": "6cb2ef64-27dd-4d7a-e549-68b7e070d6df"
      },
      "execution_count": 51,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Directory exists. Contents:\n",
            "['best-epoch=4-val_loss=-1.18.ckpt', 'last.ckpt']\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "from hashlib import sha1\n",
        "\n",
        "experiment_name = \"pretraining_lag_llama\"\n",
        "seed = 27049  # 使用你的 seed\n",
        "experiment_id = sha1(f\"{experiment_name}-seed-{seed}\".encode(\"utf-8\")).hexdigest()[:8]\n",
        "print(\"Your experiment_id is:\", experiment_id)\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "htlroQWboSDZ",
        "outputId": "ba74a2f0-fb7a-480f-e703-c4062344be55"
      },
      "execution_count": 52,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Your experiment_id is: 3056ea2a\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!ls experiments/results/pretraining_lag_llama/27049/lag-llama/3056ea2a/checkpoints/\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "CBhiagxCr_pP",
        "outputId": "087d0d19-61ee-409f-c525-fd7a01fbd333"
      },
      "execution_count": 55,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "'best-epoch=4-val_loss=-1.18.ckpt'   last.ckpt\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import os\n",
        "import shutil\n",
        "\n",
        "source_path = \"experiments/results/pretraining_lag_llama/27049/checkpoints/\"\n",
        "target_path = \"experiments/results/pretraining_lag_llama/27049/lag-llama/3056ea2a/checkpoints/\"\n",
        "\n",
        "if not os.path.exists(target_path):\n",
        "    os.makedirs(target_path, exist_ok=True)\n",
        "    print(f\"{target_path}\")\n",
        "\n",
        "for file_name in os.listdir(source_path):\n",
        "    src_file = os.path.join(source_path, file_name)\n",
        "    dst_file = os.path.join(target_path, file_name)\n",
        "    shutil.move(src_file, dst_file)\n",
        "    print(f\"{src_file} -> {dst_file}\")\n",
        "\n"
      ],
      "metadata": {
        "id": "jSk0TDo_sGIy"
      },
      "execution_count": 64,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "from data.data_utils import create_test_dataset\n",
        "\n",
        "dataset_path = \"datasets\"\n",
        "window_size = 1124 + 30\n",
        "test_data, prediction_length, total_points = create_test_dataset(\"weather\", dataset_path, window_size)\n",
        "\n",
        "print(f\"Weather dataset total points: {total_points}\")\n",
        "print(f\"Required window size: {window_size}\")"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3sCQ3grUsU-n",
        "outputId": "a94e93a3-f5f3-4f61-f37f-f11c318eadc9"
      },
      "execution_count": 57,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Weather dataset total points: 3563840\n",
            "Required window size: 1154\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "Ntqe1rXgs1un"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}